### PR TITLE
Don't mark AB as succeeded if it is not deployed

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Check that the output of the build matches what is expected from the ArtifactBuild objects, and fail the AB if the build did not produce this artifact.

I noticed that jackson-databind was missing its info, but being lumped into the jackson-core build and marked as a success.

### CODE REVIEW

File: pkg/reconciler/dependencybuild/dependencybuild.go [hunk 1]
```diff
@@ -582,6 +582,7 @@ func (r *ReconcileDependencyBuild) handleBuildPipelineRunReceived(ctx context.Co
 				} else if i.Name == artifactbuild.DeployedResources {
 					//we need to create 'DeployedArtifact' resources for the objects that were deployed
 					deployed := strings.Split(i.Value, ",")
+					db.Status.DeployedArtifacts = deployed
```
Comment (gabemontero) [inline pkg/reconciler/dependencybuild/dependencybuild.go:585]: a unit test to verify this field is getting set?

**Review by gabemontero — COMMENTED** [review]
just a minor unit test comment from me

File: pkg/reconciler/dependencybuild/dependencybuild.go (hunk 1 above)
Comment (stuartwdouglas) [reply pkg/reconciler/dependencybuild/dependencybuild.go:585]: Done

**Review by gabemontero — APPROVED** [review]
/lgtm

**Pull request merged**